File: neetcode/heap/Kth Largest Element In An Array.py
# https://leetcode.com/problems/kth-largest-element-in-an-array/

import heapq
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap = []
        for num in nums:
            heapq.heappush(heap, -num)
        
        answer = None
        while heap and k > 0:
            answer = heapq.heappop(heap)
            k -= 1
        return answer * -1

File: neetcode/go/heap/Kth Largest Element in an Array.go
package heap

func findKthLargest(nums []int, k int) int {
	minHeap := priorityqueue.NewWith(utils.IntComparator)

	for _, num := range nums {
		minHeap.Enqueue(num)
		if minHeap.Size() > k {
			minHeap.Dequeue()
		}
	}

	val, _ := minHeap.Peek()
	return val.(int)
}
